### ISSUE TITLE
Built-in CSS Module Support Not Injecting *.module.css styling during SSR

### ISSUE THREAD

**Issue by Lwdthe1** [vercel/next.js#10268]
# Bug report

## Describe the bug
The new built-in css module support does not work during SSR, at least not without further configuration that I'm not aware of. Once the page is loaded client side, the styles are injected for the component and everything looks as it should.

I checked the html using Postman http client and noticed that there's no `<style></style>` element injected in the `<head />` element. However, after the page loads in a browser, that `<style></style>` element is injected with the expected scoped css.

I tried the basic example from the blog post, and it worked. However, there's no example with SSR and components.

## To Reproduce
I have a custom `pages/_app.jsx` and a component somewhere in my `pages/index.jsx` page, which is server side rendered using `getInitialProps()`. The component imports its own ./comp.module.css file.

## Expected behavior
The css modules imported across my app structure should be gathered during server side rendering and injected into the `<head/>` of my html that is sent to the client. That way, when the user first sees my page, all the styling is in place.

#### What's sent down from SSR:
<img width="1440" alt="Screen Shot 2020-01-25 at 8 51 12 PM" src="https://user-images.githubusercontent.com/5778798/73129538-941fda00-3fb4-11ea-8cfd-4cb6c6fed58c.png">

#### What's show after the module's css is finally injected into the `<head/>` on the client:
<img width="1440" alt="Screen Shot 2020-01-25 at 8 51 13 PM" src="https://user-images.githubusercontent.com/5778798/73129539-941fda00-3fb4-11ea-8e78-652894125b25.png">

## System information

OS: macOS
Browser: Chrome
Version of Next.js: 9.2.1

## Additional context

This may be related to https://github.com/zeit/next.js/issues/10267
This is the feature I'm referring to: https://nextjs.org/blog/next-9-2#built-in-css-module-support-for-component-level-styles

**Comment by Lwdthe1** [comment]
I see that the CSS modules page mentions having to also use the postcss plugin to use css modules on the server side. Is that the suggested solution to my issue? 

https://github.com/css-modules/css-modules#server-side-and-static-websites

**Comment by IljaDaderko** [comment]
I think I am hitting same issue. I am using static export alongside scss modules and when I build my app for production my .module styles are not applied to the page initially. After I reload the page everything looks alright. I am not using any service worker caching or anything similar.

Everything works as expected in development mode.

**Comment by joonhocho** [comment]
having the same issue as well for both development and production mode.

**Comment by IljaDaderko** [comment]
For some reason css files are not loaded during first page access for me.

#### Initial page access (no css module files for the page, some global and modular css is loaded for _app component however)

<img width="1440" alt="Screenshot 2020-01-27 at 08 00 01" src="https://user-images.githubusercontent.com/3154053/73153117-3cc85980-40db-11ea-9e3d-c04584ff909f.png">

---

#### After I reload the page (modular page css is loaded)

<img width="1440" alt="Screenshot 2020-01-27 at 08 00 15" src="https://user-images.githubusercontent.com/3154053/73153124-4782ee80-40db-11ea-977f-dfe17f3366d8.png">

**Comment by timneutkens** [comment]
Can you provide a complete reproducible demo so that we can have a look

**Comment by IljaDaderko** [comment]
@timneutkens @Timer sorry for not providing it earlier, here it is https://github.com/IljaDaderko/next.js-10268

To recreate the issue:

1. Install npm dependencies
2. Run `yarn build` -> notice in package.json I changed build command to `next build; next export`
3. Run `yarn start`
4. When app loads click on `second` page link

This page should have blue background, but in my case it doesn't. If you reload the page that blue background is applied

**Comment by Timer** [comment]
@IljaDaderko this is because you're using experimental options. `experimental.modern` is not compatible with CSS support — if you turn off this feature, the CSS works as expected.

**Comment by Timer** [comment]
@Lwdthe1 can you please provide a reproducible demo? 🙏

**Comment by Lwdthe1** [comment]
Hey, I'm sorry for the delay, but my table is currently a bit full with interview projects, so I don't have time today to make the repro. I can shoot for tomorrow.

**Comment by Timer** [comment]
@Lwdthe1 any update?

**Comment by ramasilveyra** [comment]
Created issue https://github.com/zeit/next.js/issues/10404 for a very similar issue. Repro is https://github.com/ramasilveyra/bug-repro-next-global-css.

**Comment by mzygmunt** [comment]
I'm also noticing a problem with the injecting * .module.css to SSR header.

@timneutkens @Timer Reproduction is simple. Clone with-next-css example. Execute. Disable JS in the browser and refresh the page.

In my opinion, it would be appropriate to include CSS modules to the SSR page header, because SSR is very often used to improve user experience and speed up loading. Unfortunately, a site without styles looks tragic.

**Comment by Lwdthe1** [comment]
My apologies for never following up on this. I never figured it out and have moved away from it since.

**Comment by yanv1991** [comment]
https://github.com/vercel/next.js/issues/13058#issuecomment-637178095

**Comment by Timer** [comment]
Closing since OP no longer has the issue.

**Comment by Timer** [comment]
This should be fixed in `next@^9.4.5-canary.15`! Please upgrade and let us know.

**Comment by CrocoDillon** [comment]
Thanks @Timer, it seems to work! Is JavaScript disabled also supported? When I disable JS I don't see my custom styles but it could be that my setup is wrong or outdated.

**Comment by Timer** [comment]
No, JavaScript cannot be disabled in development.

**Comment by jeff-zhenz** [comment]
@Timer  Hey 

> This should be fixed in `next@^9.4.5-canary.15`! Please upgrade and let us know.

Hey, what has been fixed in this version specifically (CSS module can be SSR-ed from 9.4.5)? 

I am actually having a similar requirement: I have a standalone navbar and footer, in order to make them work in basically any React app I implemented them `react + scss module`, and I now used them in a nextjs app's layout. But according to nextjs official explanation 
https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css,  all the CSS module will be minified into `.css` files, however, what I want to have is all that CSS can be injected as inline styles during SSR (somehow like what webpack style-loader does (https://webpack.js.org/loaders/style-loader/) ) so that when the user first sees my page, all the styling is in place.

So is using style-loader in webpack in next.config.js a workaround? Any thoughts, thanks
